### ISSUE TITLE
Random Crashing

### ISSUE THREAD

**Issue by Luzr** [sirWest/AudioSwitch#10]
Any way to find out why AudoSwitch randomly crashes? It works for 4-7 hours and 3 or 4 switches but crashes.

**Comment by sirWest** [comment]
It has happened to me a few times also. Do you use remote desktop or teamviewer or other remote screen sharing software? There might be a connection when the sound devices are being remoted between computers where the host (destination) is running AudioSwitch.

**Comment by Luzr** [comment]
Yes I use RDP and Team Viewer. What do you think is the best way to narrow it down and see if that is the issue?

**Comment by sirWest** [comment]
Well in that case it most probably is the problem right there. I somehow think that it's more RDP's fault than TeamViewer's, since I've been only using TeamViewer lately and it hasn't happened anymore for me, it was frequent with RDP. Run AudioSwitch and then connect to this PC with RDP, see if the icon disappears from the tray when hovering on it with your mouse. If so, then I have to look into this on my own system with the debugger.

**Comment by Luzr** [comment]
I was able to connect in to a PC a few times with RDP and it didn't crash (I even switched the output when connected). I do have two audio cards, integrated and on board, and it looks like it crashes after a few hours. I'll let you know if I can reliability reproduce the crashing. 

Running Windows 8.1 x64.

**Comment by Luzr** [comment]
I've left Team Viewer closed for the last two days its hasn't crashed. I've connected to one Server though RDP for about 10 minutes. Maybe its TV and RDP that AudioSwitch doesn't like.

**Comment by sirWest** [comment]
RDP hands over audio sessions on Windows audio service level, TeamViewer has its own software on top of Windows. That's why I'm suspecting that it goes boom somewhere in the process of handing over or getting back the sound session(s). If the crash happens again then try to find something about it from the event viewer.

**Comment by sirWest** [comment]
has this occurred  with the latest version?

**Comment by beppe9000** [comment]
I just experienced that. I installed Audio Switch today to try it as a replacement for Audio Router (unfortunately Audio Switch lacks a lot of features compared to that, although I see better UI craftsmanship). When I used Microsoft Remote Desktop to take control of the active session (=without logout) from my android device I was greeted with a standard .Net Unexpected Exception dialog.
